### PR TITLE
Fix EvalVisitor Neq

### PR DESCRIPTION
**Pull request by MSlucgen**

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj [hunk 1]
```diff
@@ -52,6 +52,10 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+
+  <ItemGroup>
+    <None Remove="ExpressionTestCases\Equality.txt" />
+  </ItemGroup>
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj:58]: nit - can we remove this. (handled by line 47)

File: src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj [hunk 2]
```diff
@@ -52,6 +52,10 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+
+  <ItemGroup>
+    <None Remove="ExpressionTestCases\Equality.txt" />
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj:57]: nit: unintentional csproj change

File: src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj (hunk 1 above)
Comment (MSlucgen) [reply src/tests/Microsoft.PowerFx.Core.Tests/Microsoft.PowerFx.Core.Tests.csproj]: done

**Review by lesaltzm — APPROVED** [review]

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+>> Date(2022,2,17) = Date(2022,2,17)
+true
+
+>> Date(2022,2,17) <> Date(2022,2,17)
+false
+
+>> Date(2022,2,17) = Date(2017,4,27)
+false
+
+>> Date(2022,2,17) <> Date(2017,4,27)
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: Some Time test as well?

Comment (lesaltzm) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: Not sure if we have Guid or color functions yet, but that'd be good as well if they exist

Comment (MSlucgen) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: I just added Time tests
I'm not sure of the syntax for Guid or colors

Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: There is a Guid function ([https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-guid](https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-guid) ) to return guids. 

But it's not impl in interpreter - that could be a separate PR.

Comment (MSlucgen) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: I was debugging ColorValue and it seems not implemented too - another PR as well I guess

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+true
+
+>> 10% = 0.1
+true
+
+>> 10% <> 12%
+true
+
+>> 10% = 0.12
+false
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt]: fwiw, the root function doing the comparsion here is RuntimeHelpers.AreEqual()

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt (hunk 3 above)
Comment (MSlucgen) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Equality.txt:58]: image

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**